react-multiple-pages/my-app/src/App.tsx

```diff
@@ -0,0 +1,22 @@
+import { Dashboard } from './pages/Dashboard';
+import { Header } from './components/Header';
+
+import { Details } from './pages/Details';
+import { About } from './pages/About';
+import { Routes, Route } from 'react-router-dom';
+import { NotFound } from './pages/NotFound';
+
+export function App() {
+  return (
+    <>
+      <Header />
+      <Routes>
+        <Route path="/" element={<Header />} />
```

Comment:
For the Header to work properly, the routes displayed in its Outlet need to be nested. So you should put the other routes _inside_ this route. Like this:
```
<Route path="/" element{<Header />}>
  <Route index element={<Dashboard />} />
  ...
</Route>
```

See how the Route element with the Header wraps the ones inside it, so that those are its children.